ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -68,35 +76,69 @@ object EndpointsCompanion {
 
   private[http] def format(a: JsValue): ByteString = ByteString(a.compactPrint)
 
-  private[http] def decodeAndParsePayload(
-      jwt: Jwt,
-      decodeJwt: ValidateJwt): Unauthorized \/ (jwt.type, JwtPayload) =
+  private[http] def decodeAndParsePayload[A](jwt: Jwt, decodeJwt: ValidateJwt)(
+      implicit parse: ParsePayload[A]): Unauthorized \/ (jwt.type, A) = {
     for {
       a <- decodeJwt(jwt): Unauthorized \/ DecodedJwt[String]
-      p <- parsePayload(a)
+      p <- parse.parsePayload(a)
     } yield (jwt, p)
+  }
+
+  object JwtPayloadInstances {
```

Comment:
Instances of a coherent typeclass should be placed directly within a relevant companion, if possible, and then never explicitly imported for use.

Reply:
So a `ParsePayload` companion? Not quite sure how you are suggesting to remove the import from the use sites.

Reply:
Yes and yes. What's more, you have already defined the `ParsePayload` companion, so that is the best place to put them.

In this case, other places you could put the instances with the same effect would be `JwtPayload` and `JwtWritePayload`, each with its "own" instance, but that's more for when you want the instances and the typeclass itself separately defined.

Reply:
5de002e9502e9a2d03693690f7404089f23c2ad4 Thanks for the explanation!